### PR TITLE
provider/vsphere: Fix missing ssh connection info

### PR DESCRIPTION
**Pull request by tkak**
This fixes missing ssh connection info.

References: 

* https://github.com/hashicorp/terraform/issues/4112

### CODE REVIEW

**Comment by jen20** [comment]
@chrislovecnm does this look correct to you? In order to use any provisioners the connection info will need setting - it's somewhat unclear from the chain in the linked issue whether they ever worked or not?

**Comment by chrislovecnm** [comment]
@jen20 We are running into an api issue, and it would be good to get some feedback from the api owners.  I am going to file a bug with the vmware guys and see if we can loop them in.

@tkak we need to determine why we are not getting the ip addresses from mvm.Guest.Net should we do a wait for ip before?

**Comment by chrislovecnm** [comment]
Filed https://github.com/vmware/govmomi/issues/405 to the api owners.  Thanks @tkak for this, and it would be good to get `Guest.Net` to work correctly.

**Comment by chrislovecnm** [comment]
@tkak see the issue on the govmomi API. We may want to do a 'WaitForIP' on the call to get 'Guest.Net'

**Comment by lesaux** [comment]
wasn't this what the "boot_delay" parameter was supposed to deal with? This is something we had implemented a long time before the merge with this repo, but I can't find any traces of this feature anymore.

**Comment by chrislovecnm** [comment]
@lesaux not really. We need to wait for the IP address to be bound to the nic, as the owners of govmomi recommended this pattern.

Do you have concerns about this aproach?

File: builtin/providers/vsphere/resource_vsphere_virtual_machine.go [hunk 1]
```diff
@@ -484,6 +483,16 @@ func resourceVSphereVirtualMachineRead(d *schema.ResourceData, meta interface{})
 		return fmt.Errorf("Invalid network interfaces to set: %#v", networkInterfaces)
 	}
 
+	ip, err := vm.WaitForIP(context.TODO())
```
Comment (chrislovecnm) [inline builtin/providers/vsphere/resource_vsphere_virtual_machine.go]: @tkak shouldn't we be doing this earlier in the method? If we move it to before `collector.RetrieveOne` we may get better results.  The thought that I have is that we are setting the network interfaces from `Guest.Net` and the data may not be populated until `vm.WaitForIP` returns.  I am pushing a change to a branch on my repo, and maybe someone can test.

@ryanl-ee have you tested this change on this pull?  I am still looking for a test bed unfortunately, but I will make the change that I mentioned to @tkak

**Comment by chrislovecnm** [comment]
@phinze / @jen20 this has been open FOREVER, and I think we have multiple users running into this. The only thought i had was moving the code up a bit more ...  How can we get a merge??

**Comment by chrislovecnm** [comment]
@phinze this is ok for now, but I will probably tweak it later

This is a P1

**Comment by stack72** [comment]
Rebased this in #6293 - will be merging that

**Pull request closed without merging**